src/main/java/hudson/plugins/openid/OpenIdSsoSecurityRealm.java

```diff
@@ -160,12 +160,16 @@ public Authentication authenticate(Authentication authentication) throws Authent
     /**
      * The login process starts from here.
      */
-    public HttpResponse doCommenceLogin(@QueryParameter String from) throws IOException, OpenIDException {
+    public HttpResponse doCommenceLogin(@QueryParameter String from) throws IOException, OpenIDException, IllegalStateException {
         if (from == null || !from.startsWith("/")) {
             if (Stapler.getCurrentRequest().getHeader("Referer") != null) {
                 from = Stapler.getCurrentRequest().getHeader("Referer");
             } else {
-                from = Jenkins.getInstance().getRootUrl();
+                Jenkins jenkins = Jenkins.getInstance();
+                if(jenkins == null){
+                    throw new IllegalStateException("No Jenkins instance has been found.");
+                }
```

Comment:
I'm still aware about this copy-paster in multiple files, Better to just use a restricted utility method